$$\LARGE\text{Week 2 Homework}$$
$$\\[-1.5em]$$
$$\large\text{Qianlang Chen}$$
$$\\[-1.5em]$$
$$\text{Math 4400 Spring 2022}$$
$\text{}$

# Exercise 1

## Part 1

#### (a)

$$
\begin{aligned}
  & \gcd(1084, 412)\hspace{1in} & 1084 & = 2 \times 412 + 260\\
  = & \gcd(412, 260) & 412 & = 1 \times 260 + 152\\
  = & \gcd(260, 152) & 260 & = 1 \times 152 + 108\\
  = & \gcd(152, 108) & 152 & = 1 \times 108 + 44\\
  = & \gcd(108, 44) & 108 & = 2 \times 44 + 20\\
  = & \gcd(44, 20) & 44 & = 2 \times 20 + 4\\
  = & \gcd(20, 4) & 20 & = 5 \times 4 + 0\\
  = & \gcd(4, 0)\\
  = & \boxed{4}
\end{aligned}
$$

$\pagebreak$

#### (b)

$$
\begin{aligned}
  & \gcd(1979, 531)\hspace{1in} & 1979 & = 3 \times 531 + 386\\
  = & \gcd(531, 386) & 531 & = 1 \times 386 + 145\\
  = & \gcd(386, 145) & 386 & = 2 \times 145 + 96\\
  = & \gcd(145, 96) & 145 & = 1 \times 96 + 49\\
  = & \gcd(96, 49) & 96 & = 1 \times 49 + 47\\
  = & \gcd(49, 47) & 49 & = 1 \times 47 + 2\\
  = & \gcd(47, 2) & 47 & = 23 \times 2 + 1\\
  = & \gcd(2, 1) & 2 & = 2 \times 1 + 0\\
  = & \gcd(1, 0)\\
  = & \boxed{1}
\end{aligned}
$$

#### (c)

$$
\begin{aligned}
  & \gcd(305, 185)\hspace{1in} & 305 & = 1 \times 185 + 120\\
  = & \gcd(185, 120) & 185 & = 1 \times 120 + 65\\
  = & \gcd(120, 65) & 120 & = 1 \times 65 + 55\\
  = & \gcd(65, 55) & 65 & = 1 \times 55 + 10\\
  = & \gcd(55, 10) & 55 & = 5 \times 10 + 5\\
  = & \gcd(10, 5) & 10 & = 2 \times 5 + 0\\
  = & \gcd(5, 0)\\
  = & \boxed{5}
\end{aligned}
$$

In [31]:
a = 427
b = 259

print(f'  & \\gcd({a}, {b})\\hspace{{1in}}', end='')
while b > 0:
    q = a // b
    r = a % b
    print(f' & {a} & = {q} \\times {b} + {r}\\\\')
    a = b
    b = r
    print(f'  = & \\gcd({a}, {b})', end='')
print(f'\\\\\n  = & \\boxed{{{a}}}')

  & \gcd(427, 259)\hspace{1in} & 427 & = 1 \times 259 + 168\\
  = & \gcd(259, 168) & 259 & = 1 \times 168 + 91\\
  = & \gcd(168, 91) & 168 & = 1 \times 91 + 77\\
  = & \gcd(91, 77) & 91 & = 1 \times 77 + 14\\
  = & \gcd(77, 14) & 77 & = 5 \times 14 + 7\\
  = & \gcd(14, 7) & 14 & = 2 \times 7 + 0\\
  = & \gcd(7, 0)\\
  = & \boxed{7}


$\pagebreak$

## Part 2

#### (a)

$$\dfrac{1084}{412} = [2; 1, 1, 1, 2, 2, 5] = 2 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{2 + \cfrac{1}{2 + \cfrac{1}{5}}}}}}$$

#### (b)

$$\dfrac{1979}{531} = [3; 1, 2, 1, 1, 1, 23, 2] = 3 + \cfrac{1}{1 + \cfrac{1}{2 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{23 + \cfrac{1}{2}}}}}}}$$

#### (c)

$$\dfrac{305}{185} = [1; 1, 1, 1, 5, 2] = 1 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{5 + \cfrac{1}{2}}}}}$$

In [29]:
a = 305
b = 185

print(f'$$\\dfrac{{{a}}}{{{b}}}', end='')
alpha = []
while b > 0:
    q = a // b
    r = a % b
    alpha.append(q)
    a = b
    b = r

def format_alpha(alpha):
    if len(alpha) == 1: return str(alpha[0])
    return f'{alpha[0]} + \\cfrac{{{1}}}{{{format_alpha(alpha[1:])}}}'
    
print(f' = {alpha} = {format_alpha(alpha)}$$')

$$\dfrac{305}{185} = [1, 1, 1, 1, 5, 2] = 1 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{1 + \cfrac{1}{5 + \cfrac{1}{2}}}}}$$


## Part 3

#### (a)

$\dfrac{305}{185} = [1; 1, 1, 1, 5, 2]$, and $[1; 1, 1, 1, 5] = \dfrac{28}{17}$. Note that $x = 17, y = -28$ is one solution.

The other integer solutions are in the form of
$$
\begin{aligned}
  x & = 17 + \dfrac{185}{\gcd(305, 185)}k = \boxed{17 - 37k},\\
  y & = -28 + \dfrac{305}{\gcd(305, 185)}k = \boxed{-28 + 61k}
\end{aligned}
$$
for $k \in \mathbb{Z}$.

#### (b)

$\dfrac{1979}{531} = [3; 1, 2, 1, 1, 1, 23, 2]$, and $[3; 1, 2, 1, 1, 1, 23] = \dfrac{969}{260}$. Note that $x = 260, y = -969$ is one solution.

The other integer solutions are in the form of
$$
\begin{aligned}
  x & = 260 + \dfrac{1979}{\gcd(1979, 531)}k = \boxed{260 - 531k},\\
  y & = -969 + \dfrac{531}{\gcd(1979, 531)}k = \boxed{-969 + 1979k}
\end{aligned}
$$
for $k \in \mathbb{Z}$.

#### (c)

First, let's use the Euclidean Algorithm to find $\gcd(15750, 9150)$:
$$
\begin{aligned}
  & \gcd(15750, 9150)\hspace{1in} & 15750 & = 1 \times 9150 + 6600\\
  = & \gcd(9150, 6600) & 9150 & = 1 \times 6600 + 2550\\
  = & \gcd(6600, 2550) & 6600 & = 2 \times 2550 + 1500\\
  = & \gcd(2550, 1500) & 2550 & = 1 \times 1500 + 1050\\
  = & \gcd(1500, 1050) & 1500 & = 1 \times 1050 + 450\\
  = & \gcd(1050, 450) & 1050 & = 2 \times 450 + 150\\
  = & \gcd(450, 150) & 450 & = 3 \times 150 + 0\\
  = & \gcd(150, 0)\\
  = & \boxed{150}
\end{aligned}
$$

$\dfrac{15750}{9150} = [1; 1, 2, 1, 1, 2, 3]$, and $[1; 1, 2, 1, 1, 2] = \dfrac{31}{18}$. Note that $x = -18, y = -31$ is one solution.

The other integer solutions are in the form of
$$
\begin{aligned}
  x & = -18 + \dfrac{9150}{\gcd(15750, 9150)}k = \boxed{-18 - 61k},\\
  y & = 31 + \dfrac{15750}{\gcd(15750, 9150)}k = \boxed{31 + 105k}
\end{aligned}
$$
for $k \in \mathbb{Z}$.

#### (d)

First, let's use the Euclidean Algorithm to find $\gcd(427, 259)$:
$$
\begin{aligned}
  & \gcd(427, 259)\hspace{1in} & 427 & = 1 \times 259 + 168\\
  = & \gcd(259, 168) & 259 & = 1 \times 168 + 91\\
  = & \gcd(168, 91) & 168 & = 1 \times 91 + 77\\
  = & \gcd(91, 77) & 91 & = 1 \times 77 + 14\\
  = & \gcd(77, 14) & 77 & = 5 \times 14 + 7\\
  = & \gcd(14, 7) & 14 & = 2 \times 7 + 0\\
  = & \gcd(7, 0)\\
  = & \boxed{7}
\end{aligned}
$$

Since the right-hand side 13 is not divisible by 7, this equation has no integer solutions.

$\pagebreak$

# Exercise 2

## Part 1

Let's run the Continued Fraction Algorithm to see what $\sqrt{3}$ looks like as a continued fraction. Set $\alpha_0 = \sqrt{3}$.
  
  * $a_0 = \lfloor\sqrt{3}\rfloor = 1, \beta_0 = \sqrt{3} - 1, \alpha_1 = \frac{1}{\sqrt{3} - 1} = \frac{\sqrt{3} + 1}{2}$
  
  * $a_1 = \lfloor\frac{\sqrt{3} + 1}{2}\rfloor = 1, \beta_1 = \frac{\sqrt{3} + 1}{2} - 1 = \frac{\sqrt{3} - 1}{2}, \alpha_2 = \frac{2}{\sqrt{3} - 1} = \sqrt{3} + 1$
  
  * $a_2 = \lfloor\sqrt{3} + 1\rfloor = 2, \beta_2 = \sqrt{3} + 1 - 2 = \sqrt{3} - 1$

We've found a repeat as now $\beta_2 = \beta_0$. Therefore, this algorithm can never finish, meaning that $\sqrt{3}$ does not have a terminating continued fraction. So, $\sqrt{3}$ is not rational.

$\sqrt{3} \approx 1.732$, and the fourth convergent is $[1; 1, 2, 1] = \frac{7}{4} = 1.75$, which is correct for 2 digits (1 fractional digit).

## Part 2

Let's run the Continued Fraction Algorithm to see what $\sqrt{7}$ looks like as a continued fraction. Set $\alpha_0 = \sqrt{7}$.
  
  * $a_0 = \lfloor\sqrt{7}\rfloor = 2, \beta_0 = \sqrt{7} - 2, \alpha_1 = \frac{1}{\sqrt{7} - 2} = \frac{\sqrt{7} + 2}{3}$
  
  * $a_1 = \lfloor\frac{\sqrt{7} + 2}{3}\rfloor = 1, \beta_1 = \frac{\sqrt{7} + 2}{3} - 1 = \frac{\sqrt{7} - 1}{3}, \alpha_2 = \frac{3}{\sqrt{7} - 1} = \frac{\sqrt{7} + 1}{2}$
  
  * $a_2 = \lfloor\frac{\sqrt{7} + 1}{2}\rfloor = 1, \beta_2 = \frac{\sqrt{7} + 1}{2} - 1 = \frac{\sqrt{7} - 1}{2}, \alpha_3 = \frac{2}{\sqrt{7} - 1} = \frac{\sqrt{7} + 1}{3}$
  
  * $a_3 = \lfloor\frac{\sqrt{7} + 1}{3}\rfloor = 1, \beta_3 = \frac{\sqrt{7} + 1}{3} - 1 = \frac{\sqrt{7} - 2}{3}, \alpha_4 = \frac{3}{\sqrt{7} - 2} = \sqrt{7} + 2$
  
  * $a_4 = \lfloor\sqrt{7} + 2\rfloor = 4, \beta_4 = \sqrt{7} + 2 - 4 = \sqrt{7} - 2$

We've found a repeat as now $\beta_4 = \beta_0$. Therefore, this algorithm can never finish, meaning that $\sqrt{7}$ does not have a terminating continued fraction. So, $\sqrt{7}$ is not rational.

$\sqrt{7} \approx 2.646$, and the fourth convergent is $[2; 1, 1, 1] = \frac{8}{3} \approx 2.667$, which is correct for 2 digits (1 fractional digit).

## Part 3

The definition of $\alpha$ has a self-containing feature. In other words,
$$\alpha = 1 + \cfrac{1}{1 + \cfrac{1}{1 + \ddots}} = 1 + \frac{1}{\alpha}$$

By rearranging and solving,
$$
\begin{aligned}
  \alpha^2 = \alpha + 1\\
  \alpha^2 - \alpha - 1 = 0\\
  \implies \alpha = \frac{1 \pm \sqrt{5}}{2}
\end{aligned}
$$

However, from the original equation it looks like $\alpha > 0$. Therefore, $\alpha = \dfrac{1 + \sqrt{5}}{2}$.